tensorflow/core/ops/math_ops.cc

```diff
@@ -1447,4 +1447,52 @@ b: Another tensor, of same type and shape as `a`.
 product: Pairwise cross product of the vectors in `a` and `b`.
 )doc");
 
+// --------------------------------------------------------------------------
+
+REGISTER_OP("ScanSum")
```

Comment:
I decided to go with `scan_sum` and `scan_prod` based on this comment: https://github.com/tensorflow/tensorflow/issues/813#issuecomment-179365203
The idea was to keep it consistent with `reduce_*`.
But if the consensus is to go with `cumsum`/`cumprod` I'd be happy to change it.